confluent_kafka/avro/serializer/__init__.py

```diff
@@ -33,8 +61,270 @@ def __str__(self):
 
 
 class KeySerializerError(SerializerError):
-    pass
+    def __new__(cls, message):
+        return super(KeySerializerError, cls).__new__(SerializerError, message, True)
 
 
 class ValueSerializerError(SerializerError):
     pass
+
+
+class ContextStringIO(io.BytesIO):
```

Comment:
I always found this thing funny... you do know that there is `contextlib.closing`, right?

Reply:
Nope, although to be honest I just reused the existing code. Patches welcome though :)